test/unit/mock/readable-storage-logic.spec.ts

```diff
@@ -0,0 +1,175 @@
+      expect(await mock.getAddress()).to.equal(ADDRESS_EXAMPLE);
+
+      const getValue = await mock.getVariable('_address');
+      const expectedValue = await mock.getAddress();
+      expect(getValue).to.equal(expectedValue);
+    });
+
+    it('should be able to get a boolean', async () => {
+      await mock.setVariable('_bool', true);
+      expect(await mock.getBool()).to.equal(true);
+
+      const getValue = await mock.getVariable('_bool');
+      const expectedValue = await mock.getBool();
+      expect(getValue).to.equal(expectedValue);
+    });
+
+    it('should be able to get a small value < 31 bytes', async () => {
+      const value = BYTES_EXAMPLE.slice(0, 10);
+      await mock.setVariable('_bytes', value);
+      expect(await mock.getBytes()).to.equal(value);
+
+      const getValue = await mock.getVariable('_bytes');
+      const expectedValue = await mock.getBytes();
+      expect(getValue).to.equal(expectedValue);
+    });
+
+    it('should be able to get a large value > 31 bytes', async () => {
+      await mock.setVariable('_bytes', BYTES_EXAMPLE);
+      expect(await mock.getBytes()).to.equal(BYTES_EXAMPLE);
+
+      const getValue = await mock.getVariable('_bytes');
+      const expectedValue = await mock.getBytes();
+      expect(getValue).to.equal(expectedValue);
+    });
+
+    it('should be able to get bytes32', async () => {
+      await mock.setVariable('_bytes32', BYTES32_EXAMPLE);
+      expect(await mock.getBytes32()).to.equal(BYTES32_EXAMPLE);
+
+      const getValue = await mock.getVariable('_bytes32');
+      const expectedValue = await mock.getBytes32();
+      expect(getValue).to.equal(expectedValue);
+    });
+
+    it('should be able to set a simple struct', async () => {
+      const struct = {
+        valueA: BigNumber.from(1234),
+        valueB: true,
+      };
+      await mock.setVariable('_simpleStruct', struct);
+
+      expect(convertStructToPojo(await mock.getSimpleStruct())).to.deep.equal(struct);
+
+      const getValue = await mock.getVariable('_simpleStruct');
+      expect(getValue).to.deep.equal(struct);
+    });
+
+    it('should be able to get an address in a packed struct', async () => {
+      const struct = {
+        packedA: true,
+        packedB: ADDRESS_EXAMPLE,
+      };
+      await mock.setVariable('_packedStruct', struct);
+
+      expect(convertStructToPojo(await mock.getPackedStruct())).to.deep.equal(struct);
+
+      const getValue = await mock.getVariable('_packedStruct');
+      expect(getValue).to.deep.equal(struct);
+    });
+
+    it('should be able to get an address in a packed struct', async () => {
+      const struct = {
+        packedA: BigNumber.from(2),
+        packedB: BigNumber.from(1),
+        packedC: BigNumber.from(2),
+        packedD: BigNumber.from(1),
+        packedE: ADDRESS_EXAMPLE,
+      };
+      await mock.setVariable('_packedStruct2', struct);
+      const getValue = await mock.getVariable('_packedStruct2');
```

Comment:
hmmm i'd generalize the `packedStruct` so that we don't have 2, and use the same for multiple reasons